src/server/campaignd/server.cpp

```diff
@@ -220,6 +222,73 @@ void server::load_config()
 	// Since an addon is sent in a single WML document this essentially limits
 	// the maximum size of an addon that can be uploaded.
 	simple_wml::document::document_size_limit = cfg_["document_size_limit"].to_int(default_document_size_limit);
+
+	//Loading addons
+	addons_.clear();
+	std::vector<std::string> legacy_addons, dirs;
+	filesystem::get_files_in_dir("data", &legacy_addons, &dirs);
+	config meta;
+	for(const std::string& addon_dir : dirs) {
+		in = filesystem::istream_file(filesystem::normalize_path("data/" + addon_dir + "/addon.cfg"));
+		read(meta, *in);
+		if(meta) {
+			addons_.emplace(meta["name"].str(), meta);
+		} else {
+			WRN_CS << "Failed to load addon from dir '" << addon_dir << "'\n";
```

Comment:
The server needs to die here instead of issuing a warning that absolutely nobody will see. If this condition is reached odds are that something is horribly wrong with the filesystem or its contents. Proceeding in this state is bound to result in data or metadata loss later.